stream_alert/shared/lookup_tables/driver_s3.py

```diff
@@ -0,0 +1,199 @@
+from datetime import datetime, timedelta
+import json
+import time
+import sys
+import zlib
+
+import boto3
+from botocore.exceptions import ClientError, ConnectTimeoutError, ReadTimeoutError
+
+import stream_alert.shared.helpers.boto as boto_helpers
+from stream_alert.shared.logger import get_logger
+from stream_alert.shared.lookup_tables.drivers import PersistenceDriver
+from stream_alert.shared.lookup_tables.errors import LookupTablesInitializationError
+
+LOGGER = get_logger(__name__)
+
+
+class S3Driver(PersistenceDriver):
+    """
+    S3Driver
+
+    This PersistenceDriver uses AWS S3 as the backing layer. The entire table is stored in a single
+    large S3 file.
+
+    Upon initialization, the S3 file is loaded, decompressed, decoded, and loaded into memory as
+    a Python dict.
+
+    To improve performance, the decoded contents of the S3 file are cached in-memory for up to
+    "cache_refresh_minutes" in minutes of time, after which they are reloaded from S3.
+
+    (!) NOTE: Tweaking the cache_refresh_minutes will likely not help memory performance of your
+        LookupTable. If you are having memory issues on your Lambda, try using the DynamoDBDriver.
+    """
+    _DEFAULT_CACHE_REFRESH_MINUTES = 10
+
+    def __init__(self, configuration):
+        # Example configuration:
+        # {
+        #     "driver": "s3",
+        #     "bucket": "airbnb.sample.lookuptable",
+        #     "key": "resource_map.gz",
+        #     "cache_refresh_minutes": 10,
+        #     "compression": "gzip",
+        #     "key_delimiter": "|"
+        # },
+
+        super(S3Driver, self).__init__(configuration)
+
+        self._s3_bucket = configuration['bucket']
+        self._s3_key = configuration['key']
+        self._compression = configuration.get('compression', False)
+        self._cache_refresh_minutes = configuration.get(
+            'cache_refresh_minutes',
+            self._DEFAULT_CACHE_REFRESH_MINUTES
+        )
+
+        self._last_load_time = None
+
+        self._s3_data = {}
+        self._dirty = False
```

Comment:
can you leave a comment as to what the dirty attribute is for

Reply:
My bad; it's gonna be for using the `.set()` method in the future.... currently not used. Will remove until I get to that future commit